L1Trigger/L1THGCal/src/HGCalTriggerTowerGeometryHelper.cc

```diff
@@ -84,35 +82,67 @@ HGCalTriggerTowerGeometryHelper::HGCalTriggerTowerGeometryHelper(const edm::Para
   }
 
   if (splitModuleSum_){
+    //variables for transforming towers
+    rotate180Deg_ = int(nBinsPhi_)/2; 
+    rotate120Deg_ = int(nBinsPhi_)/3;
+    reverseX_ = int(nBinsPhi_)/2 - 1;
+
     std::ifstream moduleTowerMappingStream(conf.getParameter<edm::FileInPath>("moduleTowerMapping").fullPath());
     if (!moduleTowerMappingStream.is_open()) {
       throw cms::Exception("MissingDataFile") << "Cannot open HGCalTowerMapProducer moduleTowerMapping file\n";
     }
+    //get split divisors
     std::string line;
-    std::getline(moduleTowerMappingStream, line); //To skip first row (header) of the file
+    std::getline(moduleTowerMappingStream, line); //Skip row
+    std::getline(moduleTowerMappingStream, line); 
+    std::stringstream ss(line);
+    ss>>splitDivisorSilic_>>splitDivisorScint_;
+    
+    //get towers and module sum shares
+    std::getline(moduleTowerMappingStream, line); //Skip row
+    std::getline(moduleTowerMappingStream, line); //Skip row
+    const int minNumOfWordsPerRow = 5;
+    const int numOfWordsPerTower = 3;
     for( std::string line; std::getline(moduleTowerMappingStream, line ); ){
-        std::vector<std::string> lineSplitted;
-        std::stringstream ss(line);
-        while(ss.good()){
-            std::string substr;
-            std::getline(ss, substr, ' ');
-            lineSplitted.push_back(substr);
+      int NumOfWordsInThisRow = 0;
```

Comment:
Variable names should not start with a capital letter (usually reserved for class names).
So should rather be `numOfWordsInThisRow`

Reply:
> Thanks a lot @mhassans I just have one remaining minor naming comment below. Once implemented, could you:
> 
> * rebase your work on top of the latest devel branch (currently `CMSSW_12_0_0_pre3`). Given that you have a large number of commits, it might be simpler to squash them first and rebase.
> * run `scram b code-format` and commit the proposed changes

Thanks @jbsauvan 

I implemented the comment and then squashed the commits. Could you please confirm if the rebasing can be done in the following way?

Create a branch from the latest CMSSW version:
`git checkout -b my_CMSSW_12_0_0_pre3 pfcaldev/hgc-tpg-devel-CMSSW_12_0_0_pre3`
Then update it with origin (pfcaldev):
`git pull pfcaldev hgc-tpg-devel-CMSSW_12_0_0_pre3`
Back to my branch:
`git checkout moh-modSplit`
Finally rebase:
`git rebase my_CMSSW_12_0_0_pre3`
Then force push it to my branch:
`git push myRemote +moh-modSplit`

Reply:
Hi @mhassans 
That looks correct. In case you want to check that the rebase will pick the commit you want, you can do an interactive rebase (with `-i`)

Reply:
<img width="1036" alt="Screen Shot 2021-11-02 at 12 16 34 PM" src="https://user-images.githubusercontent.com/22216879/139844556-63e75c9e-f726-4218-b9de-c8556a30ed5a.png">

@jbsauvan 
There are some merging conflicts after I did the rebase (please see the attached screenshot). Should I resolve it by choosing lines from the newest release (12_0_0_pre3), or the conflicts are not expected?

Reply:
There should be conflict only in the files that you modified, which doesn't seem to be the case here. So these conflicts are not expected. I'll have a look.

Reply:
It seems that the rebase wants to apply a bunch of commits unrelated to this PR. The simplest then is probably to `cherry-pick` your single commit. Or you can still do the rebase but do it interactively (with `-i`) and remove all the commits except your commit.

Reply:
By the way, I just modified the base branch of the PR to the latest devel branch (this is why there are now conflicts). When you push the rebased branch, it should be able to merge again.

Reply:
@jbsauvan Thanks! The rebasing went well this time and I also did code-format check. So please let me know about the next step if everything looks fine.